source/MaterialXCore/Document.cpp

```diff
@@ -169,9 +170,12 @@ void Document::importLibrary(const ConstDocumentPtr& library, const CopyOptions*
         {
             childCopy->setGeomPrefix(library->getGeomPrefix());
         }
-        if (!childCopy->hasColorSpace() && library->hasColorSpace())
+        if (!libraryColorSpace.empty())
         {
-            childCopy->setColorSpace(library->getColorSpace());
+            if (!childCopy->hasColorSpace() || childCopy->isA<Output>())
```

Comment:
An output's final color space should be the documents colorspace if it was specified. This information was being "lost".